sw/device/tests/crypto/cryptotest/firmware/hmac.c

```diff
@@ -97,16 +94,51 @@ status_t handle_hmac(ujson_t *uj) {
       .len = tag_bytes / sizeof(uint32_t),
       .data = tag_buf,
   };
+  // Test oneshot API
   otcrypto_status_t status = otcrypto_hmac(&key, input_message, tag);
   if (status.value != kOtcryptoStatusValueOk) {
     return INTERNAL(status.value);
   }
-  // Copy tag to uJSON type
+  // Copy oneshot tag to uJSON type
   cryptotest_hmac_tag_t uj_tag;
-  memcpy(uj_tag.tag, tag_buf, tag_bytes);
+  memcpy(uj_tag.oneshot_tag, tag_buf, tag_bytes);
   uj_tag.tag_len = tag_bytes;
 
-  // Send tag to host via UART
+  // Zero out tag_buf to mitigate chance of a false positive in the
+  // stepwise test
+  memset(tag_buf, 0, tag_bytes);
+  // Test streaming API
+  otcrypto_hmac_context_t ctx;
+  status = otcrypto_hmac_init(&ctx, &key);
+  if (status.value != kOtcryptoStatusValueOk) {
+    return INTERNAL(status.value);
+  }
+  // Split up input message into 2 shares for better coverage of
+  // streaming API
+  otcrypto_const_byte_buf_t input_message_share1 = {
+      .len = uj_message.message_len / 2,
+      .data = msg_buf,
+  };
+  otcrypto_const_byte_buf_t input_message_share2 = {
+      .len = ceil_div(uj_message.message_len, 2),
+      .data = &msg_buf[uj_message.message_len / 2],
```

Comment:
nit: it's equivalent, but maybe clearer to write:
```suggestion
      .len = uj_message.message_len - input_message_share1.len,
      .data = &msg_buf[input_message_share1.len],
```
This way, you could even change the first share to e.g. `message_len / 3` and not have to change the second share at all; it removes a certain dependency between them.